src/frontends/common/src/frontend.cpp

```diff
@@ -74,6 +77,7 @@ void FrontEnd::normalize(const std::shared_ptr<Model>& model) const {
 
 void FrontEnd::add_extension(const std::shared_ptr<ov::Extension>& extension) {
     if (m_actual) {
+        m_shared_objects.push_back(extension);
```

Comment:
maybe we should add `OPENVINO_ASSERT(!m_shared_objects.empty())` here to ensure that extension will not be added as first item (where plugin SO pointer should be)